os_windows_test.go

```diff
@@ -0,0 +1,82 @@
+// +build windows
+
+package ctxvfs
+
+import (
+	"testing"
+)
+
+func TestWindowsMount(t *testing.T) {
+
+	// keys are mount points,
+	// values contain expected results for osFS.resolve(key)
+	tests := map[string]map[string]string{
+		"": {
+			"C:/folder/file":  "C:\\folder\\file",
+			"D:/folder/file":  "D:\\folder\\file",
+			"/C:/folder/file": "C:\\folder\\file",
+			"/folder/file":    "",
+			"file":            "",
+			"/file":           "",
+		},
+		"C:": {
+			"C:/folder/file":  "C:\\folder\\file",
+			"/C:/folder/file": "C:\\folder\\file",
```

Comment:
> why are you allowed to specify C:/ and /C:/. Surely you should only be able to specify paths relative to C:? I expect the above to cases to resolve to the empty string.

I believe we should allow anything that located under the given mount point when combined with `root`. With the old implementation on Unix we can resolve "/foo/bar" to /"foo/bar" with mount point defined as a "/foo" so why do it different way on Windows?

>  Does the old vfs.OS("C:") have this behaviour? 

It doesn't. However on Unix old implementation gives the following results for `osFS.resolve` using `/foo` as a mount point:

"/foo/bar" =>  "/foo/bar", "bar =>  "/foo/bar"

I think that we should support both absolute and relative paths while they still located under a given mount point (or located system-wide in the cases of `/` or `""`)